From 5b59075ec02032f7ac7dcafe26bf649d68c01947 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 23 Dec 2017 21:30:54 +0000
Subject: [linux-steam-integration 03/11] Revert "intercept: Allow disabling
 (build-time) the udev redirection"

This reverts commit c640f2f215b4636ed99cfa4f807b38dfc25f526c.

Turns out this is due to a very weird situation where Arch has their
libudev.so.0 as an incompatible API which all current games expect to
be of libudev.so.1 ABI, and isn't something LSI should be fixing, the
distro should.
---
 meson.build          | 7 -------
 meson_options.txt    | 2 --
 src/intercept/main.c | 4 ----
 3 files changed, 13 deletions(-)

diff --git a/meson.build b/meson.build
index fe9415e..1b45f5c 100644
--- a/meson.build
+++ b/meson.build
@@ -90,13 +90,6 @@ if with_snap_support == true
     message('Enabling snapd support. Do NOT do this for distribution builds!')
 endif
 
-with_udev_redirect = get_option('with-udev-redirect')
-if with_udev_redirect == true
-    cdata.set('HAVE_UDEV_REDIRECT', '1')
-    message('Enabling udev soname redirection. DO NOT DO THIS FOR ARCH LINUX!')
-    message('The udev redirect should only be used when not using udev shim')
-endif
-
 config_h = configure_file(
      configuration: cdata,
      output: 'config.h',
diff --git a/meson_options.txt b/meson_options.txt
index ea6798e..f915cca 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -14,5 +14,3 @@ option('with-libressl-suffix', type: 'string', value: '-libressl',
         description: 'Suffix for shim LibreSSL library when using --with-libressl-mode=shim')
 
 option('with-snap-support', type: 'boolean', description: 'Build LSI with explicit support for snapd', value: false)
-
-option('with-udev-redirect', type: 'boolean', description: 'Enable libudev soname redirection. DO NOT USE WITH UDEV SHIM.')
diff --git a/src/intercept/main.c b/src/intercept/main.c
index 392454b..9f6e4c0 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -273,9 +273,7 @@ static const char *vendor_transmute_source[] = {
 
         "libbz2.so.1.0",
 
-#ifdef HAVE_UDEV_REDIRECT
         "libudev.so.0",
-#endif
 };
 
 /**
@@ -313,9 +311,7 @@ static const char *vendor_transmute_target[] = {
 
         "libbz2.so.1.0.6",
 
-#ifdef HAVE_UDEV_REDIRECT
         "libudev.so.1",
-#endif
 };
 
 /**
-- 
2.20.0

